cpp/src/io/parquet/experimental/hybrid_scan_helpers.hpp

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "io/parquet/parquet_gpu.hpp"
+#include "io/parquet/reader_impl_helpers.hpp"
+
+#include <cudf/io/detail/utils.hpp>
+#include <cudf/io/experimental/hybrid_scan.hpp>
+#include <cudf/io/types.hpp>
+#include <cudf/utilities/memory_resource.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/mr/device/device_memory_resource.hpp>
+
+#include <memory>
+#include <optional>
+#include <vector>
+
+namespace cudf::io::parquet::experimental::detail {
+
+using aggregate_reader_metadata_base = parquet::detail::aggregate_reader_metadata;
+using equality_literals_collector    = parquet::detail::equality_literals_collector;
+using input_column_info              = parquet::detail::input_column_info;
+using metadata_base                  = parquet::detail::metadata;
+using row_group_info                 = parquet::detail::row_group_info;
+
+/**
+ * @brief Class for parsing dataset metadata
+ */
+struct metadata : private metadata_base {
+  explicit metadata(cudf::host_span<uint8_t const> footer_bytes);
+  metadata_base get_file_metadata() && { return std::move(*this); }
+};
+
+class aggregate_reader_metadata : public aggregate_reader_metadata_base {
```

Comment:
Derived from `aggregate_reader_metadata` in `reader_impl_helpers.hpp` and extends its functionality.